ext/crt.c

```diff
@@ -41,34 +43,49 @@ static PHP_GINIT_FUNCTION(awscrt) {
     awscrt_globals->log_level = 0;
 }
 
-/* awscrt_version */
-ZEND_BEGIN_ARG_INFO(awscrt_version_arginfo, 0)
-ZEND_END_ARG_INFO()
+/* aws_crt_last_error() */
+PHP_FUNCTION(aws_crt_last_error) {
+    RETURN_LONG(aws_crt_last_error());
+}
+
+/* aws_crt_error_str(int error_code) */
+PHP_FUNCTION(aws_crt_error_str) {
+    zend_ulong error_code = 0;
 
-PHP_FUNCTION(awscrt_version) {
-    static const char *version = "1.0.0-dev";
-    AWS_RETURN_STRING(version);
+    if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "l", &error_code) == FAILURE) {
+        RETURN_NULL();
+    }
+
+    AWS_RETURN_STRING(aws_crt_error_str(error_code));
 }
 
-/* aws_crt_last_error() */
-ZEND_BEGIN_ARG_INFO(aws_crt_last_error_arginfo, 0)
-ZEND_END_ARG_INFO()
+/* aws_crt_error_name(int error_code) */
+PHP_FUNCTION(aws_crt_error_name) {
+    zend_ulong error_code = 0;
 
-PHP_FUNCTION(aws_crt_last_error) {
-    RETURN_LONG(aws_crt_last_error());
+    if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "l", &error_code) == FAILURE) {
+        RETURN_NULL();
+    }
+
+    AWS_RETURN_STRING(aws_crt_error_name(error_code));
 }
 
-/* clang-format off */
-const zend_function_entry awscrt_functions[] = {
-    PHP_FE(awscrt_version, awscrt_version_arginfo)
-    PHP_FE(aws_crt_last_error, aws_crt_last_error_arginfo)
-    PHP_FE_END
-};
+/* aws_crt_error_debug_str(int error_code) */
+PHP_FUNCTION(aws_crt_error_debug_str) {
+    zend_ulong error_code = 0;
 
+    if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "l", &error_code) == FAILURE) {
+        RETURN_NULL();
+    }
+
+    AWS_RETURN_STRING(aws_crt_error_debug_str(error_code));
+}
+
+/* clang-format off */
```

Comment:
should be able to add a ',' to the last struct member and not need to disable format?

Reply:
OH RIGHT, lemme try that